### PR TITLE
Changes to WireMock reset location

### PR DESCRIPTION
**Pull request by seran**

### CODE REVIEW

File: core/src/main/kotlin/org/evomaster/core/search/service/SearchAlgorithm.kt [hunk 1]
```diff
@@ -90,6 +94,8 @@ abstract class SearchAlgorithm<T> where T : Individual {
 
         ssu.enabled = false
 
+        externalServiceHandler.resetWireMockServers()
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/search/service/SearchAlgorithm.kt:97]: hmmm... from an architectural point of view, no sure we should have reference to WM in `SearchAlgorithm`. would need to discuss where best to put it

**Review by arcuri82 — CHANGES_REQUESTED** [review]

**Comment by arcuri82** [comment]
@seran conflicts after merging other PRs into master

File: core/src/main/kotlin/org/evomaster/core/search/service/SearchAlgorithm.kt (hunk 1 above)
Comment (seran) [reply core/src/main/kotlin/org/evomaster/core/search/service/SearchAlgorithm.kt:97]: can we move the `Minimizer` from `SearchAlgorithm` to the `Main.initAndRun`?

Comment (arcuri82) [reply core/src/main/kotlin/org/evomaster/core/search/service/SearchAlgorithm.kt:97]: hmmm... in theory could be possible, but i would prefer not.

why would be important to reset WM before minimization? doesn't each test case using WM resetting its instances anyway?

Comment (seran) [reply core/src/main/kotlin/org/evomaster/core/search/service/SearchAlgorithm.kt:97]: I did this just to make sure WM is clean before `Minimizer`. Let me disable the change and see whether it works or breaks.

**Review by arcuri82 — APPROVED** [review]

**Pull request merged**